ergotree-ir/src/sigma_protocol/dlog_group.rs

```diff
@@ -115,37 +116,30 @@ pub fn random_scalar_in_group_range() -> Scalar {
     Scalar::generate_vartime(&mut OsRng)
 }
 
-/// Attempts to create BigInt from Scalar
-pub fn scalar_to_bigint(s: Scalar) -> BigInt {
+/// Attempts to create BigInt256 from Scalar
+/// Returns None if s > 2^255 - 1
```

Comment:
```suggestion
/// Returns None if s > 2^255 - 1
Since Scalar is in [0, n) range, where n is the group order (FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFE BAAEDCE6 AF48A03B BFD25E8C D0364141) it might not fit into 256-bit BigInt because BigInt uses 1 bit for sign.
```

Reply:
Added a comment